Assets/Scripts/Editor/BeatMappingWindow.cs

```diff
@@ -0,0 +1,198 @@
+
+            _soundClipField.objectType = typeof(AudioClip);
+            rootVisualElement.hierarchy.Add(_soundClipField);
+        }
+
+        /// <summary>
+        /// Creates basic audio control interface.
+        /// </summary>
+        private void CreateAudioControlInterface()
+        {
+            _volumeSlider = new Slider("Volume Slider", 0, 1);
+            _volumeSlider.value = _waveformDrawer.audioSource.volume;
+
+            _scrubAmoundField = new FloatField("Time Scrub Amount");
+            _scrubAmoundField.value = _defaultTimeScrubAmount;
+            _waveformDrawer.timeScrubAmount = _scrubAmoundField.value;
+
+            _keyBindsButton = new Button(() => { _waveformDrawer.LogKeybinds(); });
+            _keyBindsButton.text = "Keybinds";
+
+            _importTimeStampsButton = new Button(() => { _timeStamper.TryImportTimeStamps(); });
+            _importTimeStampsButton.text = "Import Time Stamps";
+
+            rootVisualElement.Add(_volumeSlider);
+            rootVisualElement.Add(_scrubAmoundField);
+            
+            rootVisualElement.Add(_keyBindsButton);
+            rootVisualElement.Add(_importTimeStampsButton);
+        }
+
+        /// <summary>
+        /// Callback method which is called when user clicks on one of the song names in the list view.
+        /// </summary>
+        /// <param name="obj">Selected audioclip</param>
+        private void OnSongIndexChanged(IEnumerable<object> obj)
+        {
+            List<object> objectList = obj.ToList();
+            if (objectList.Count == 0) //check if selected song is not empty
+                return;
+
+            AudioClip audioClip = (AudioClip)objectList[0];
+            if (audioClip == null) //check if cast to audio clip was valid
+                return;
+
+            _soundClipField.value = audioClip;
+            if (Application.isPlaying)
+            {
+                if (_spectrumDrawer != null) //tell spectrum data to draw the waveform of the selected audio clip
+                    _spectrumDrawer.VisualizeSongSpectrum(audioClip);
+            }
+            else
+            {
+                PlayModeWarning();
+            }
+        }
+
+        /// <summary>
+        /// Method which warns the user if they try to use this window not in playmode.
+        /// </summary>
+        private void PlayModeWarning()
+        {
+            if (!Application.isPlaying)
+                Debug.LogWarning("Enter play-mode to use BeatMapping Tool");
+        }
+
+        private void Update()
+        {
+            //update audio control values
+            if (Application.isPlaying && _waveformDrawer != null)
+            {
+                _waveformDrawer.audioSource.volume = _volumeSlider.value;
+                _waveformDrawer.timeScrubAmount = _scrubAmoundField.value;
+            }
+            else
+            {
+                Close(); //close window if not in playmode
+            }
```

Comment:
Remove else and return in the if statement above.
```suggestion
            // Close window if not in playmode.
            Close();
```